great_expectations/cli/batch_request.py

```diff
@@ -424,9 +424,7 @@ def _get_batch_spec_passthrough(
     elif isinstance(datasource, SimpleSqlalchemyDatasource):
         # Some backends require named temporary table parameters.  We specifically elicit those and add them
         # where appropriate.
-        execution_engine: SqlAlchemyExecutionEngine = cast(
-            SqlAlchemyExecutionEngine, datasource.execution_engine
-        )
+        datasource.execution_engine
```

Comment:
This never gets used.
Should we remove this line too? Or is there a reason to ensure that this attribute exists?

Reply:
This entire method seems like it could be removed in favor of a single error check in the `else`.

Reply:
I want to say we keep it as some sort of check to make sure that property works as intended? we can probably get away from the cast though.

Maybe git blame and let's discuss with whoever wrote it?

Reply:
@alexsherstinsky 
Although I'd prefer to save runtime changes to another PR (unless there's actually a bug or misuse).

Reply:
@Kilo59 This is very weird and looks like you found a bug.  It says that I authored that line, but GitBlame has been inconsistent in my recent reviews (so I am not sure how this bug ended up in there).  Regardless, could you please try deleting that line that references `datasource.execution_engine` entirely to see if it will work without it?  Thanks!

Reply:
@alexsherstinsky I really don't want to get into changing run time behavior in this PR.
This was meant to be a quick PR to bring some of this CLI code under type coverage.